### DISCUSSION TITLE
How to configure redirect after standard actions?

### DISCUSSION THREAD

**Discussion by whysthatso** [janko/rodauth-rails#184] [Q&A] [answered]
I'm embedding forms for 'change password' and 'change login' in a profile model's show view. after submitting successful changes to either password or login i am redirected to the root page.

how would i change this behaviour, i.e. redirect back to the profile show action?

**Comment by janko** [comment, marked as answer]
There are `*_redirect` configuration methods for various redirects, so in this case you could do:

```rb
change_password_redirect { rails_routes.profile_path } # or just "/profile"
change_login_redirect { rails_routes.profile_path } # or just "/profile"
```

There is also a `default_redirect` which most redirects default to, you can change it if you'd like most actions to the same place.

**Reply by whysthatso** [reply]
that's a really  specification, and possibly worth an addition to this part of the documentation:  https://github.com/janko/rodauth-rails#rails-url-helpers

i have a hard time understanding what exactly is available in this context.

to explain, i have a profile very much like the one in the rodauth-demo-rails app, so i thought i would do a redirect along:

```rb
change_password_redirect { rails_routes.profile_path(Profile.find_by!(account_id: account_id)) }
```

It works, but referencing the Profile model like so, is that the best way to do this or is there a preferred idiom for that?

**Reply by janko** [reply]
> that's a really specification, and possibly worth an addition to this part of the documentation: https://github.com/janko/rodauth-rails#rails-url-helpers

Sounds reasonable, added ✅ 

> i have a hard time understanding what exactly is available in this context.

The feature documentation lists all existing configuration methods, so if you open e.g. [confirm_password docs](http://rodauth.jeremyevans.net/rdoc/files/doc/confirm_password_rdoc.html), you'll see `confirm_password_redirect` there.

> It works, but referencing the Profile model like so, is that the best way to do this or is there a preferred idiom for that?

Given that the logged in user has only one profile, I would recommend making it a singular resource in your routes, i.e. `resource :profile`. Then you'd retrieve the current profile via `current_account.profile`.
